### PR TITLE
Add function to scrape Video IDs from YouTube Search Resource.

### PR DESCRIPTION
**Pull request by josefj1519**

### CODE REVIEW

File: src/test/java/com/google/step/youtube/YouTubeInfoScraperTest.java [hunk 1]
```diff
@@ -58,6 +62,8 @@
     private static final String UPLOAD_ID = "UPLOAD_ID";
     private static final String VIDEO_ID = "VIDEO_ID";
     private static final String VIDEO_TITLE = "VIDEO_TITLE";
+    private static final String KEYWORD = "KEYWORD";
+    private static final String KEYWORD_NO_RESULTS = "KEYWORD_NO_RESULTS";
```
Comment (dallen68) [inline src/test/java/com/google/step/youtube/YouTubeInfoScraperTest.java]: Change to NO_RESULTS_KEYWORD to stay consistent with putting adjective before noun.

**Review by dallen68 — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/youtube/YouTubeInfoScraper.java [hunk 2]
```diff
@@ -103,6 +105,25 @@ public YouTubeInfoScraper() {
         return Optional.of(response.getItems());
     }
 
+    /**
+     * @param keyword Word to search with.
+     * @return an optional list of videoIds.The optional will be empty if id is
```
Comment (margaretchan) [inline src/main/java/com/google/step/youtube/YouTubeInfoScraper.java]: Doesn't really matter but you're missing a space between the sentences.

File: src/main/java/com/google/step/youtube/YouTubeInfoScraper.java [hunk 3]
```diff
@@ -103,6 +105,25 @@ public YouTubeInfoScraper() {
...
+     *         invalid.
+     */
+    public Optional<List<String>> scrapeVideoIdsFromSearch(String keyword) throws IOException {
+        SearchListResponse response = youTubeClient.search().list("").setMaxResults(50L).setQ(keyword)
+                .setFields("items(id)").execute();
+        if (response.getItems() == null) {
+            return Optional.empty();
+        }
+        checkState(!response.getItems().isEmpty(), "Expected more than 0 SearchResult items to be found.");
```
Comment (margaretchan) [inline src/main/java/com/google/step/youtube/YouTubeInfoScraper.java:117]: What if the query is a keyboard smash and there are no results found? We should probably still return `Optional.empty()` in this case.

**Review by margaretchan — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/step/youtube/YouTubeInfoScraper.java (hunk 3 above)
Comment (josefj1519) [reply src/main/java/com/google/step/youtube/YouTubeInfoScraper.java:117]: The API will return null if empty.  We are just checking for impossible scenarios.

**Review by margaretchan — APPROVED** [review]

**Review by dallen68 — APPROVED** [review]

File: src/main/java/com/google/step/youtube/YouTubeInfoScraper.java [hunk 4]
```diff
@@ -103,6 +105,25 @@ public YouTubeInfoScraper() {
         return Optional.of(response.getItems());
     }
 
+    /**
+     * @param keyword Word to search with.
+     * @return an optional list of videoIds. The optional will be empty if id is
+     *         invalid.
+     */
+    public Optional<List<String>> scrapeVideoIdsFromSearch(String keyword) throws IOException {
+        SearchListResponse response = youTubeClient.search().list("").setMaxResults(50L).setQ(keyword)
```
Comment (LiRq95) [inline src/main/java/com/google/step/youtube/YouTubeInfoScraper.java]: Move 50L to constant

**Review by LiRq95 — CHANGES_REQUESTED** [review]

**Review by LiRq95 — APPROVED** [review]

**Pull request merged**